CMakeLists.txt

```diff
@@ -30,35 +30,95 @@ if(NOT DEFINED CMAKE_INSTALL_CMAKEDIR)
+
 if(nanopb_BUILD_GENERATOR)
-    set(generator_protos nanopb plugin)
-
-    find_package(PythonInterp 2.7 REQUIRED)
-    execute_process(
-        COMMAND ${PYTHON_EXECUTABLE} -c
-            "from distutils import sysconfig; print(sysconfig.get_python_lib(prefix='${CMAKE_INSTALL_PREFIX}'))"
-        OUTPUT_VARIABLE PYTHON_INSTDIR
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-    )
+    set(python_versions 2.7 3.4 3.5 3.6 3.7)
+    set(python_instdirs "")
+	foreach(python_version IN LISTS python_versions)
+        set(python_exe "")
+        string(REPLACE "." ";" python_version_list ${python_version})
+		list(GET python_version_list 0 python_version_major)
+        set(python_possibles python${python_version} python${python_version_major} python)
+        foreach(python_possible IN LISTS python_possibles)
+            execute_process(
+                COMMAND ${python_possible} -c
+                    "import sys; print('.'.join([str(x) for x in sys.version_info[:2]]))"
+                OUTPUT_VARIABLE python_version_actual
+                OUTPUT_STRIP_TRAILING_WHITESPACE
+            )
+            if(python_version EQUAL python_version_actual)
+                set(python_exe ${python_possible})
+                break()
+            endif()
+        endforeach()
+        if(python_exe)
+            message(STATUS "Found python: " ${python_exe})
+            execute_process(
+                COMMAND ${python_exe} -c
+                    "from distutils import sysconfig; print(sysconfig.get_python_lib(prefix='${CMAKE_INSTALL_PREFIX}'))"
+                OUTPUT_VARIABLE python_instdir
+                OUTPUT_STRIP_TRAILING_WHITESPACE
+            )
+            list(APPEND python_instdirs ${python_instdir})
+			message(STATUS "Installing generator libraries to: " ${python_instdir})
+        endif()
+    endforeach()
+
+    set(PROTOBUF_DIR "" CACHE STRING "Location of the protobuf directory")
+    set(PROTOBUF_INC_DIR "" CACHE STRING "Location of the protobuf include directory")
+    if (NOT "${PROTOBUF_INC_DIR}" STREQUAL "")
+        set(protoc_FLAGS "-I${PROTOBUF_INC_DIR}")
+    elseif(NOT "${PROTOBUF_DIR}" STREQUAL "")
+        set(protoc_FLAGS "-I${PROTOBUF_DIR}/include")
+    else()
+        include(FindPkgConfig)
+        pkg_search_module(protobuf protobuf)
+        if(protobuf_FOUND)
+            string(REGEX REPLACE "([^;]+)" "-I\\1" protoc_FLAGS "${protobuf_INCLUDE_DIRS}")
+        else()
+            set(protoc_FLAGS "-I/usr/include")
+        endif()
+    endif()
+    message(STATUS "Protoc Flags: " ${protoc_FLAGS})
 
     foreach(generator_proto IN LISTS generator_protos)
-        string(REGEX REPLACE "([^;]+)" "generator/proto/\\1.proto" generator_proto_file "${generator_proto}")
-        string(REGEX REPLACE "([^;]+)" "\\1_pb2.py" generator_proto_py_file "${generator_proto}")
+        string(REGEX REPLACE "([^;]+)" "${CMAKE_SOURCE_DIR}/generator/nanopb/\\1.proto" generator_proto_file "${generator_proto}")
```

Comment:
${PROJECT_SOURCE_DIR} is a better option. If you have a top-level CMakeLists.txt that includes this one in a add_subdirectory() call, this logic will break because the resulting path will be rooted at the top-level CMakeList's folder.